### ISSUE TITLE
Should CPR be weighted by distance climbed, or at least take it into consideration?

### ISSUE THREAD

**Issue by brendanheywood** [theCrag/website#2938]
Split off from #1698 around CPR for multipitch. One thing we should aim to retain is cpr score consistency. There are 3 scenarios:

a) I simply onsight a 200m route with 10 pitches of the same grade
b) I entered onsight for every pitch and said I lead each pitch
c) the route was instead entered as 10 individual routes and I onsighted them all

c in particular is quite important as logging 10 routes vs 1 is enough to push you up a whole extra grade in cpr value. If I was to alternate leads then, or different tick types for different grades, etc then this can get really hairy. I can see two options:

1) we weight cpr by length, and each pitch gets its own cpr which then all gets added up to the ascent cpr. a == b == c

2) we calculate the cpr for each pitch, and then choose the maximum to be the overall ascent cpr score. a == b but b != c

So 1) is more correct but it's potential a can of worms as it's a fairly big change and should really also apply to all the single pitch routes too, and for simplicity probably everything across the board, boulder dws ice etc. It also has the potential to add slightly more meaning to comparing cpr's for boulders directly to routes. It still won't be great at comparing say an easy route with a hard crux vs a consistently hard route. And we also run into semantic subtleties where some grade systems in theory are the grade of the hardest moves while others in theory are an overall route grade. Lastly we have to deal with route missing length data, so we need a sane fallback in the algorithm like 20m for routes and 3m for boulders.

We will always have to contend with partially incorrect ascent data where someone says 'jim led the crux not me' but the overall grade wasn't changed, I think this scenario will create dud data in equally bad ways for both algorithm so doesn't help decide.

My pragmatic gut says to just take the easy option and go with the max option. But my inner nerd says to weight by length. I am also very curious to see how much impact this would have on cpr for single pitch sport routes, and if it radically changes any of the scores or ranks at the top end of town (or bottom end).

This perhaps also touches on some other areas, for instance I've had quite a few people mention that they felt like 'trad climbers' and not sport or boulder, but their profile says otherwise because it's working off their ascent count and not ascent total length. Increasingly we are leaning on cpr for more metrics on the profile and elsewhere so there are benefits to this being consistently applied.

**Comment by scd** [comment]
Doing 10 pitches of all the same crux grade is no easy feat, so I would be happy giving it the higher CPR. Luckily this is very rare as most multipitch routes will have not more than two routes at the crux grade. 

There is a third algorithm you missed for comparison.

I think the algorithm we want is sum of CPRs for each pitch, assuming tick types were logged at each pitch. This would mean logging as 10 separate ticks would give you the same CPR as logging one ascent with all pitches. 

I don't think we need to add length in at all. The fundamental problem I have with length is that a grade 21 route may be 35 meters of grade 18 and 5 meters of grade 21. Is that really much harder then a 10 meter grade 21 climb. 

I therefore think our atomic unit is pitch, and that our choice is to go with max of pitch CPRs or sum of pitch CPRs. I think for choosing the overall tick type of the ascent (ie second or lead) we use max but for the assigned CPR we use sum.

The sum of pitch CPRs would give an interesting objective view of how hard the overall route is and be a valuable metric in it's own right.

We could easily look at the effect of this over the 20 most climbed multi pitches over 3 pitches and compare algorithms.

I am in no rush with this. Choosing the overall tick type is something that needs to be resolved as part of the issue for logging pitch tick types but adjusting the CPR algorithm is not something we need to do immediately.

**Comment by brendanheywood** [comment]
Using pitch is really just a coarse substitute for length anyway, the difference is that grades can take into account the pitch length. So ok to rule out length in lieu of pitches.

On a long trad route the number of pitches is often fairly arbitrary and you could stretch it out on 70m ropes. Doing a 200m in 3 x 66m pitches should at least be equal, or really be worth more, than a 200m in 10 x 20m pitches which has more rests. If we use sum this is the wrong way round, but if we use max then they are at least worth the same. That is an edge case, but it's the same problem with just ticking a 200m with no pitches at all which would be very penalized by using sum (but I'm ok with slightly penalizing people who don't enter more detailed info). So I'd lean just slightly towards max. I think your comment about grade consistency is what matters, very few routes will have consistent grades all the way so it won't make much difference for the vast majority of routes where the one crux pitch cpr will swamp the rest. Lets test it but it probably doesn't matter either way.

**Comment by scd** [comment]
Implementing max is no change because that is effectively what it is now and it will default to this when we auto select the overall ascent grade.

Once we have wider acceptance of the CPR system I think we should be a able to get a good discussion going and work out whether on not there is value in implementing a sum approach. Even though I prefer it, I would not want to do something like that without consultation.

As we have resolve the discussion do we keep this issue open, or close it?

**Comment by brendanheywood** [comment]
Given that there is no time pressure I should we leave this open for public comment

**Comment by Drazhar** [comment]
Imho the amount of pitches matter, but not the distance climbed.

The length of the route is already considered in the grade. A 66m route which has one hard move and the rest easy stuff, shouldn't be higher rated than a 10m route with a really hard boulder sequence but the same grade.

**Comment by rouletout** [comment]
I agree, the length of a climb is inherently in the grade so should not have an influence on the CPR.
